### PR TITLE
UX improvements to Settings

### PR DESCRIPTION
**Pull request by stefanalund**
## Hey, I just made a Pull Request!

Screenshot 2020-10-17 at 20 35 16
Screenshot 2020-10-17 at 20 35 31
Screenshot 2020-10-17 at 20 40 27

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/spotify/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)

### CODE REVIEW

**Comment by freben** [comment]
I always thought, shouldn't the Pin Sidebar toggle also be a switch like the ones used for feature flags? I don't think it needs its own iconography.

File: plugins/user-settings/src/components/General/General.tsx [hunk 1]
```diff
@@ -21,12 +21,12 @@ import { Profile } from './Profile';
 import { ThemeToggle } from './ThemeToggle';
 
 export const General = () => (
-  <Grid container spacing={3}>
-    <Grid item md={12}>
+  <Grid container direction="row" spacing={3}>
+    <Grid item sm={6}>
```
Comment (freben) [inline plugins/user-settings/src/components/General/General.tsx:25]: In general when making grid items, do try them out in a smaller window size and add like `md={12}` or similar to make them stack vertically on constrained spaces

**Review by freben — APPROVED** [review]

**Comment by stefanalund** [comment]
@freben thanks -- fixed both things.

**Comment by stefanalund** [comment]
@freben any suggestions how to make the test work?

**Comment by stefanalund** [comment]
Maybe some help in here: https://github.com/mui-org/material-ui/issues/17697

**Comment by freben** [comment]
@stefanalund it's `data-testid`, not `data-test-id`

**Comment by freben** [comment]
@stefanalund Maybe it could be interesting to also try to use `userEvent.click` instead of `fireEvent`. It really works a lot more like an actual user and browser does, by emitting the entire series of events that may or may not be relevant. Like mouse enters the thing, mouse moves a bit, button is pressed down, button is released ..." instead of just the low level "click" and nothing else. This sometimes works a lot better. We don't use it a lot yet but it feels like the future :)

https://github.com/testing-library/user-event

File: plugins/user-settings/src/components/General/PinButton.test.tsx [hunk 2]
```diff
@@ -32,11 +32,10 @@ describe('<PinButton />', () => {
         </SidebarPinStateContext.Provider>,
       ),
     );
-
     expect(rendered.getByText('Pin Sidebar')).toBeInTheDocument();
 
-    const pinButton = rendered.getByTitle('Pin Sidebar');
-    fireEvent.click(pinButton);
+    const pinButton = rendered.getByTestId('pin');
+    fireEvent.change(pinButton);
```
Comment (abhishekjakhar) [inline plugins/user-settings/src/components/General/PinButton.test.tsx:38]:
```suggestion
    const pinButton = rendered.getByLabelText('Pin Sidebar Switch');
    fireEvent.click(pinButton);
```

**Review by abhishekjakhar — COMMENTED** [review]
This will make the test case work. When we use debug method provided by RTL we are able to see the DOM which our component is making when it renders. I have attached a screenshot which tells us the DOM. Here you can see the input is having `aria-label` of `Pin Sidebar Switch`. I tried running the test using these changes and it is now working for me.

testing-library

**Review by freben — APPROVED** [review]

**Comment by stefanalund** [comment]
Thanks for the help @abhishekjakhar — much appreciated!

**Comment by abhishekjakhar** [comment]
> Thanks for the help @abhishekjakhar — much appreciated!

Glad to help.

**Pull request merged**